### PR TITLE
refactor: more edge case tests for CLIP audio encoder

### PR DESCRIPTION
**Pull request by bwanglzu**
1. simplification of readme
2. more edge test cases.

Part of #84

### CODE REVIEW

File: jinahub/encoders/audio/AudioCLIPEncoder/README.md [hunk 1]
```diff
@@ -1,8 +1,7 @@
 # AudioCLIPEncoder
 
-**AudioCLIPEncoder** is a class that wraps the [AudioCLIP](https://github.com/AndreyGuzhov/AudioCLIP) model for generating embeddings for audio data. 
+**AudioCLIPEncoder** is a class that wraps the [AudioCLIP](https://github.com/AndreyGuzhov/AudioCLIP) model for generating embeddings for audio data.
 
+This encoder is meant to be used in conjunction with the [AudioCLIPImageEncoder](https://github.com/jina-ai/executors/tree/main/jinahub/encoders/image/AudioCLIPImageEncoder) and [AudioCLIPTextEncoder](https://github.com/jina-ai/executors/tree/main/jinahub/encoders/text/AudioCLIPTextEncoder), as it can embedd text, images and audio to the same latent space.
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/README.md:5]:
```suggestion
This encoder is meant to be used in conjunction with the [AudioCLIPTextEncoder](https://hub.jina.ai/executor/jfe8kovq) and [AudioCLIPTextEncoder](https://hub.jina.ai/executor/jfe8kovq), as it can embed text, images and audio to the same latent space.
```

File: jinahub/encoders/audio/AudioCLIPEncoder/README.md [hunk 2]
```diff
@@ -1,8 +1,7 @@
...
 
-## Reference
-- [AudioCLIP paper](https://arxiv.org/abs/2106.13043)
-- [AudioCLIP code](https://github.com/AndreyGuzhov/AudioCLIP)
+For more information, such as run executor on gpu, check out [documentation](https://docs.jina.ai/tutorials/gpu-executor/).
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/README.md:7]:
```suggestion
For more information, such as run executor on gpu, check out [documentation](https://docs.jina.ai/tutorials/gpu-executor/).

## See Also
- [AudioCLIPTextEncoder](https://hub.jina.ai/executor/jfe8kovq)
- [AudioCLIPImageEncoder](https://hub.jina.ai/executor/3atsazub)
```

**Review by nan-wang — CHANGES_REQUESTED** [review]

**Comment by nan-wang** [comment]
@bwanglzu I suggest adding `batch_size` for this to enable batching. Check out the codes in `CLIPTextEncoder`.

**Comment by nan-wang** [comment]
BTW, is the model downloading part mentioned in the README? I agree with that we don't download models by default. Do you throw a warning about this

File: jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import torch
 from jina import DocumentArray, Executor, requests
 from jina.excepts import BadDocType
+from jina_commons.batching import get_docs_batch_generator
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py:11]:
```suggestion
```

File: jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py [hunk 4]
```diff
@@ -58,25 +66,19 @@ def encode(
...
-        flat_docs = docs.traverse_flat(traversal_paths)
-
-        # filter out documents without audio wav
-        filtered_docs = DocumentArray(
-            [doc for doc in flat_docs if doc.blob is not None]
+        batch_generator = get_docs_batch_generator(
+            docs,
+            traversal_path=parameters.get('traversal_paths', self.traversal_paths),
+            batch_size=parameters.get('batch_size', self.batch_size),
+            needs_attr='blob',
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py:76]:
```suggestion
traversal_path=parameters.get('traversal_paths', self.traversal_paths)
batch_size=parameters.get('batch_size', self.batch_size)
```

File: jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py [hunk 5]
```diff
@@ -58,25 +66,19 @@ def encode(
...
         )
 
-        return filtered_docs
+        with torch.no_grad():
+            for batch in batch_generator:
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py:80]:
```suggestion
            for batch in docs.batch(batch_size, traversal_paths, required_attr='blob'):
```

File: jinahub/encoders/audio/AudioCLIPEncoder/gpu_requirements.txt [hunk 6]
```diff
@@ -8,4 +8,5 @@ visdom==0.1.8.9
 ftfy==6.0.3
 librosa==0.8.0
 regex==2021.7.6
-ffmpeg
\ No newline at end of file
+ffmpeg
+jina-commons @ git+https://github.com/jina-ai/jina-commons.git#egg=jina-commons
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/gpu_requirements.txt:12]:
```suggestion
```

File: jinahub/encoders/audio/AudioCLIPEncoder/requirements.txt [hunk 7]
```diff
@@ -1,6 +1,7 @@
 -f https://download.pytorch.org/whl/torch_stable.html
 torch==1.9.0+cpu
 torchvision
+jina-commons @ git+https://github.com/jina-ai/jina-commons.git#egg=jina-commons
```
Comment (nan-wang) [inline jinahub/encoders/audio/AudioCLIPEncoder/requirements.txt:4]:
```suggestion
```

**Review by nan-wang — CHANGES_REQUESTED** [review]
FYI, https://docs.jina.ai/api/jina.types.arrays.traversable/?highlight=batch#jina.types.arrays.traversable.TraversableSequence.batch

File: jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py (hunk 5 above)
Comment (bwanglzu) [reply jinahub/encoders/audio/AudioCLIPEncoder/executor/audio_clip_encoder.py:80]: ah i didn't even know we have batch already , great!

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Review by nan-wang — APPROVED** [review]
LGTM👍
